integration/base/src/main/java/com/inrupt/client/integration/base/AccessGrantScenarios.java

```diff
@@ -236,252 +235,254 @@ static void teardown() {
-    void accessGrantGetRdfTest(final Session session) {
+    void accessGrantGetRdfTest(final Session resourceOwnerSession, final Session requesterSession) {
         LOGGER.info("Integration Test - Fetching RDF using Access Grant");
 
-        final SolidSyncClient client = SolidSyncClient.getClientBuilder()
-            .build().session(session);
+        final SolidSyncClient resourceOwnerClient = SolidSyncClient.getClientBuilder()
+            .build().session(resourceOwnerSession);
 
         try (final SolidRDFSource resource = new SolidRDFSource(testRDFresourceURI, null, null)) {
-            assertDoesNotThrow(() -> client.create(resource));
+            assertDoesNotThrow(() -> resourceOwnerClient.create(resource));
+
+            prepareACPofResource(resourceOwnerClient, testRDFresourceURI, SolidRDFSource.class);
         }
 
-        final AccessGrantClient accessGrantClient = new AccessGrantClient(
+        final AccessGrantClient requesterAccessGrantClient = new AccessGrantClient(
                 URI.create(ACCESS_GRANT_PROVIDER)
-        ).session(session);
+        ).session(requesterSession);
 
         final Set<String> modes = new HashSet<>(Arrays.asList(GRANT_MODE_READ, GRANT_MODE_WRITE, GRANT_MODE_APPEND));
         final Instant expiration = Instant.parse(GRANT_EXPIRATION);
-        final AccessRequest request = accessGrantClient.requestAccess(URI.create(webidUrl),
+        final AccessRequest request = requesterAccessGrantClient.requestAccess(URI.create(requesterWebidUrl),
             new HashSet<>(Arrays.asList(testRDFresourceURI)), modes, PURPOSES, expiration)
             .toCompletableFuture().join();
-        final AccessGrant grant = accessGrantClient.grantAccess(request)
+
+        final AccessGrantClient resourceOwnerAccessGrantClient = new AccessGrantClient(
+            URI.create(ACCESS_GRANT_PROVIDER)
+        ).session(resourceOwnerSession);
+        final AccessGrant grant = resourceOwnerAccessGrantClient.grantAccess(request)
             .toCompletableFuture().join();
-        final Session newSession = AccessGrantSession.ofAccessGrant(session, grant);
-        final SolidSyncClient newClient = SolidSyncClient.getClientBuilder()
+        final Session newSession = AccessGrantSession.ofAccessGrant(requesterSession, grant);
+        final SolidSyncClient requesterClient = SolidSyncClient.getClientBuilder()
             .build().session(newSession);
 
-        try (final SolidRDFSource resource = newClient.read(testRDFresourceURI, SolidRDFSource.class)) {
+        try (final SolidRDFSource resource = requesterClient.read(testRDFresourceURI, SolidRDFSource.class)) {
             assertTrue(resource.getMetadata().getContentType().contains(Utils.TEXT_TURTLE));
         }
 
-        newClient.delete(testRDFresourceURI);
-        assertDoesNotThrow(accessGrantClient.revoke(grant).toCompletableFuture()::join);
+        //cleanup
+        resourceOwnerClient.delete(testRDFresourceURI);
```

Comment:
For other resources, this is done in the BeforeAll and AfterAll functions. Should it be the case for this too? This applies to other tests with RDF resources too. In particular, I'm wondering wether the current setup would properly clean up after a failed test.

Reply:
The resource is created locally reason why we delete it immediately here.